pkg/pillar/types/assignableadapters.go

```diff
@@ -455,3 +464,61 @@ func (aa *AssignableAdapters) ExpandControllers(log *base.LogObject, list []*IoB
 	}
 	return elist
 }
+
+func getDeviceNameFromPciID(pciID string) (string, error) {
+	// e.g., ls /sys/bus/pci/devices/<pciID>/nvme/
+	//  -> nvme0
+	deviceName := ""
+	nvmePath, err := ioutil.ReadDir(sysfsPciDevices + pciID + "/nvme")
+	if err != nil {
+		return "", err
+	}
+	for _, file := range nvmePath {
+		deviceName = file.Name()
+	}
+	return deviceName, nil
+}
+
+// NVMEIsUsed checks if an NVME device is in a ZFS pool or mounted
+func NVMEIsUsed(log *base.LogObject, zfsPoolStatusMap map[string]interface{}, pciID string) bool {
+	if _, err := os.Stat(zfsManagerDir); os.IsNotExist(err) {
```

Comment:
This doesn't seem like a very robust check.
Adding a Initialized boolean to the ZfsPoolStatus would be more robust, as long as we get one ZfsPoolStatus even when ZFS is not in use.
If not, one can add a check whether we are using ZFS or ext4. In pillar/vault package we have a ReadPersistType() which will tell you that.
And if ZFS is used you should have (at least) one element in zfsPollStratusMap, if not zfsmanager has not yet reported the status.